fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/PermissiveAvailabilitySpec.groovy

```diff
@@ -0,0 +1,145 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.availability
+
+import com.yahoo.bard.webservice.data.config.names.TableName
+import com.yahoo.bard.webservice.metadata.TestDataSourceMetadataService
+import com.yahoo.bard.webservice.table.Column
+import com.yahoo.bard.webservice.table.resolver.DataSourceConstraint
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList
+
+import org.joda.time.Interval
+
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import java.util.stream.Collectors
+
+class PermissiveAvailabilitySpec extends Specification  {
+
+    PermissiveAvailability permissiveAvailability
+    TableName tableName
+    Column column1
+    Column column2
+    Interval interval1
+    Interval interval2
+
+    def setup() {
+        tableName = TableName.of('table')
+
+        column1 = new Column('column1')
+        column2 = new Column('column2')
+
+        permissiveAvailability = new PermissiveAvailability(
+                tableName,
+                [column1, column2] as Set,
+                new TestDataSourceMetadataService()
+        )
+    }
+
+    def "getDataSourceNames returns the correct table name"() {
+        expect:
+        permissiveAvailability.dataSourceNames == Collections.singleton(TableName.of('table'))
+    }
+
+    def "getAllAvailability returns all availabilities for each columns configured to the table"() {
```

Comment:
I think the new test is the same as `getAvailableInterval returns only configured column`